torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -190,17 +190,14 @@ def resize_image_tensor(
         if interpolation == InterpolationMode.NEAREST or interpolation == InterpolationMode.NEAREST_EXACT:
             # uint8 dtype can be included for cpu and cuda input if nearest mode
             acceptable_dtypes.append(torch.uint8)
-        elif interpolation == InterpolationMode.BILINEAR and image.device.type == "cpu":
+        elif (
+            interpolation == InterpolationMode.BILINEAR
+            and image.is_cpu
+            and "AVX2" in torch.backends.cpu.get_cpu_capability()
+        ):
             # uint8 dtype support for bilinear mode is limited to cpu and
             # according to our benchmarks non-AVX CPUs should prefer u8->f32->interpolate->u8 path
-            if "AVX2" in torch.backends.cpu.get_cpu_capability():
```

Comment:
This could have been in the original `elif` already or am I missing something?

Reply:
yes it's logically the same